### PR TITLE
DAOS-10892 control: Add backward compatibility for enable_vmd param

### PR DESCRIPTION
**Pull request by tanabarr**
Enable the use of enable_vmd server config file parameter in order to be
backwards compatible with DAOS 2.0 config files.

The use of both enable_vmd and disable_vmd will return an error when
loading the config file during daos_server start-up.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>
Required-githooks: true

### CODE REVIEW

File: src/control/cmd/dmg/auto_test.go [hunk 1]
```diff
@@ -179,6 +179,7 @@ access_points:
 - hostX:10002
 fault_cb: ""
 hyperthreads: false
+enable_vmd: null
```
Comment (kjacque) [inline src/control/cmd/dmg/auto_test.go:182]: A little confused by this -- shouldn't this be omitted?

File: src/control/server/config/server_legacy.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+//
+
+package config
+
+// ServerLegacy describes old configuration options that should be supported for backward
+// compatibility and may be deprecated in future releases.
+// See utils/config/daos_server.yml for parameter descriptions.
+type ServerLegacy struct {
+	// Detect outdated "enable_vmd" config parameter and direct users to update config file.
+	EnableVMD *bool `yaml:"enable_vmd"`
```
Comment (kjacque) [inline src/control/server/config/server_legacy.go:14]: needs "omitempty"

File: src/control/server/config/server_legacy.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package config
+
+// ServerLegacy describes old configuration options that should be supported for backward
+// compatibility and may be deprecated in future releases.
+// See utils/config/daos_server.yml for parameter descriptions.
+type ServerLegacy struct {
```
Comment (kjacque) [inline src/control/server/config/server_legacy.go:12]: Would it make sense to move the legacy Servers param here now too?

**Review by kjacque — CHANGES_REQUESTED** [review]

File: src/control/server/config/server_legacy.go (hunk 2 above)
Comment (tanabarr) [reply src/control/server/config/server_legacy.go]: done

File: src/control/cmd/dmg/auto_test.go (hunk 1 above)
Comment (tanabarr) [reply src/control/cmd/dmg/auto_test.go]: done

**Review by knard-intel — APPROVED** [review]
OK for me

**Pull request merged**